api/filter/http_connection_manager.proto

```diff
@@ -0,0 +1,250 @@
+  // processed sequentially as request events happen.
+  repeated HttpFilter http_filters = 5;
+
+  // Whether the connection manager manipulates the user-agent and
+  // x-envoy-downstream-service-cluster headers. See the linked documentation
+  // for more information. Defaults to false.
+  google.protobuf.BoolValue add_user_agent = 6;
+
+  message Tracing {
+    // Span name will be derived from operation_name.
+    enum OperationName {
+      INGRESS = 0;
+      EGRESS = 1;
+    }
+    OperationName operation_name = 1;
+
+    // The header name is used to populate the tag name, and the header value is
+    // used to populate the tag value. The tag is created if the specified
+    // header name is present in the request’s headers.
+    repeated string request_headers_for_tags = 2;
+  }
+  Tracing tracing = 7;
+
+  // TODO(htuch): GrpcProtocolOptions?
+  oneof http_codec_options {
+    Http1ProtocolOptions http_protocol_options = 8;
+    Http2ProtocolOptions http2_protocol_options = 9;
+  }
+
+  // An optional override that the connection manager will write to the server
+  // header in responses. If not set, the default is envoy.
+  string server_name = 10;
+
+  // The idle timeout for connections managed by the connection manager. The
+  // idle timeout is defined as the period in which there are no active
+  // requests. If not set, there is no idle timeout. When the idle timeout is
+  // reached the connection will be closed. If the connection is an HTTP/2
+  // connection a drain sequence will occur prior to closing the connection. See
+  // drain_timeout.
+  google.protobuf.Duration idle_timeout = 11;
+
+  // The time that Envoy will wait between sending an HTTP/2 “shutdown
+  // notification” (GOAWAY frame with max stream ID) and a final GOAWAY frame.
+  // This is used so that Envoy provides a grace period for new streams that
+  // race with the final GOAWAY frame. During this grace period, Envoy will
+  // continue to accept new streams. After the grace period, a final GOAWAY
+  // frame is sent and Envoy will start refusing new streams. Draining occurs
+  // both when a connection hits the idle timeout or during general server
+  // draining. The default grace period is 5000 milliseconds (5 seconds) if this
+  // option is not specified.
+  google.protobuf.Duration drain_timeout = 12;
+
+  // Configuration for HTTP access logs emitted by the connection manager.
+  AccessLog access_log = 13;
+
+  // If set to true, the connection manager will use the real remote address of
+  // the client connection when determining internal versus external origin and
+  // manipulating various headers. If set to false or absent, the connection
+  // manager will use the x-forwarded-for HTTP header. See the documentation for
+  // x-forwarded-for, x-envoy-internal, and x-envoy-external-address for more
+  // information.
+  google.protobuf.BoolValue use_remote_address = 14;
```

Comment:
Can you add a TODO here to clean this up. I can work on a PR or file an issue when I'm back from vacation (or we can discuss tomorrow). I think this needs to be made more explicit in terms of what type of sanitizing is done and header setting is done. This is related to XFF, the new XFCC stuff, etc.